From 25e0c75fff1ce5c2df4ebc81a08f51586466564a Mon Sep 17 00:00:00 2001
From: Nickolay Shmyrev <nshmyrev@gmail.com>
Date: Thu, 7 Jan 2021 22:33:16 +0100
Subject: [PATCH] Properly cache arcmap

---
 src/fstext/fstext-utils-inl.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/fstext/fstext-utils-inl.h b/src/fstext/fstext-utils-inl.h
index 44e641a3f20..7d491a17559 100644
--- a/src/fstext/fstext-utils-inl.h
+++ b/src/fstext/fstext-utils-inl.h
@@ -151,9 +151,10 @@ template<class Arc, class I>
 LookaheadFst<Arc, I> *LookaheadComposeFst(const Fst<Arc> &ifst1,
                                           const Fst<Arc> &ifst2,
                                           const std::vector<I> &to_remove) {
-  fst::CacheOptions cache_opts(true, 1 << 25LL);
-  fst::CacheOptions cache_opts_map(true, 0);
-  fst::ArcMapFstOptions arcmap_opts(cache_opts);
+  fst::CacheOptions cache_opts(true, 0);
+  fst::CacheOptions cache_opts_map(true, 1 << 26LL);
+  fst::ArcMapFstOptions arcmap_opts(cache_opts_map);
+
   RemoveSomeInputSymbolsMapper<Arc, I> mapper(to_remove);
   return new LookaheadFst<Arc, I>(ComposeFst<Arc>(ifst1, ifst2, cache_opts), mapper, arcmap_opts);
 }
