diff -up kaldi-master/src/util/kaldi-thread.cc.4~ kaldi-master/src/util/kaldi-thread.cc
--- kaldi-master/src/util/kaldi-thread.cc.4~	2025-09-22 09:05:59.000000000 +0200
+++ kaldi-master/src/util/kaldi-thread.cc	2025-11-17 17:16:04.776956857 +0100
@@ -27,7 +27,7 @@ int32 g_num_threads =
 #ifdef KALDI_WASM
 0 // Everything will run on the main thread for WASM
 #else 
-8  // Initialize this global variable.
+4  // Initialize this global variable.
 #endif
 ;
 MultiThreadable::~MultiThreadable() {
