diff -up kaldi-master/tools/extras/check_dependencies.sh.3~ kaldi-master/tools/extras/check_dependencies.sh
--- kaldi-master/tools/extras/check_dependencies.sh.3~	2025-09-22 09:05:59.000000000 +0200
+++ kaldi-master/tools/extras/check_dependencies.sh	2025-11-17 17:15:03.746043053 +0100
@@ -109,43 +109,6 @@ else
   pythonok=true
 fi
 
-(
-#Use a subshell so that sourcing env.sh does not have an influence on the rest of the script
-[ -f ./env.sh ] && . ./env.sh
-rm -f $PWD/python/python*
-if ! [ -f $PWD/python/.use_default_python ]; then
-  echo "$0: Configuring python"
-  echo "$0: ... If you really want to avoid this, add an" \
-         "empty file $PWD/python/.use_default_python and run this script again."
-  if $python27 ; then
-    echo "$0: ... python2.7 found, making it default (python, python2, python2.7)"
-    ln -s $(command -v python2.7) $PWD/python/python
-    ln -s $(command -v python2.7) $PWD/python/python2
-    ln -s $(command -v python2.7) $PWD/python/python2.7
-  fi
-
-  if $python3 ; then
-    echo "$0: ... python3 found, making symlink (python3)"
-    ln -s $(command -v python3) $PWD/python/python3
-    if ! $python27 ; then
-      echo "$0: ... ... python2.7 not found, using python3 as python"
-      ln -s $(command -v python3) $PWD/python/python
-    fi
-  fi
-else
-  echo "$0: Not configuring python(s) -- using system defaults"
-  if ! have python ; then
-    echo "$0: WARNING: 'python' executable not present, configuring"
-    if $python27 ; then
-      ln -s $(command -v python2.7) $PWD/python/python
-    elif $python3 ; then
-      ln -s $(command -v python3) $PWD/python/python
-    fi
-  fi
-fi
-
-)
-
 mathlib_missing=false
 case "$(uname -m)-$(uname -s)" in
   x86_64*)  # Suggest MKL on an Intel64 system (not supported on i?86 hosts).
