diff -up kaldi-master/CMakeLists.txt.8~ kaldi-master/CMakeLists.txt
--- kaldi-master/CMakeLists.txt.8~	2025-09-22 09:05:59.000000000 +0200
+++ kaldi-master/CMakeLists.txt	2025-11-17 17:16:53.843927727 +0100
@@ -61,7 +61,7 @@ unset(IS_LIB_SHARE)
 
 if(BuildForFedora)
 else()
-    set(CMAKE_CXX_STANDARD 14)
+    set(CMAKE_CXX_STANDARD 17)
     set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 set(CMAKE_INSTALL_MESSAGE LAZY) # hide "-- Up-to-date: ..."
