diff -up kaldi-2024.05.19/CMakeLists.txt.omv~ kaldi-2024.05.19/CMakeLists.txt
--- kaldi-2024.05.19/CMakeLists.txt.omv~	2024-05-19 16:01:36.092187735 +0200
+++ kaldi-2024.05.19/CMakeLists.txt	2024-05-19 16:01:42.708261568 +0200
@@ -87,7 +87,7 @@ if(CONDA_ROOT)
     # https://github.com/kaldi-asr/kaldi/blob/master/src/matrix/kaldi-blas.h#L95-L113
     add_definitions(-DHAVE_OPENBLAS=1)
     if(MSVC)
-        link_libraries(cblas lapack)
+        link_libraries(openblas lapack)
         # necessary macros to compile on windows, from here:
         # https://icl.cs.utk.edu/lapack-for-windows/lapack/
         add_definitions(-DADD_)
diff -up kaldi-2024.05.19/cmake/gen_cmake_skeleton.py.omv~ kaldi-2024.05.19/cmake/gen_cmake_skeleton.py
--- kaldi-2024.05.19/cmake/gen_cmake_skeleton.py.omv~	2024-05-19 16:01:19.396001399 +0200
+++ kaldi-2024.05.19/cmake/gen_cmake_skeleton.py	2024-05-19 16:01:23.244044346 +0200
@@ -269,7 +269,7 @@ class CMakeListsLibrary(object):
 
         if len(self.depends) > 0:
             ret.append("target_link_libraries(" + self.target_name + " PUBLIC")
-            for d in self.depends + ['-lcblas', '-llapack']:
+            for d in self.depends + ['-lopenblas', '-llapack']:
                 ret.append("    " + d)
             ret.append(")\n")
 
