diff -up kaldi-master/cmake/gen_cmake_skeleton.py.26~ kaldi-master/cmake/gen_cmake_skeleton.py
--- kaldi-master/cmake/gen_cmake_skeleton.py.26~	2024-05-20 23:47:38.484941459 +0200
+++ kaldi-master/cmake/gen_cmake_skeleton.py	2024-05-20 23:48:18.661368376 +0200
@@ -269,7 +269,7 @@ class CMakeListsLibrary(object):
 
         if len(self.depends) > 0:
             ret.append("target_link_libraries(" + self.target_name + " PUBLIC")
-            for d in self.depends + ['-lcblas', '-llapack']:
+            for d in self.depends + ['-lopenblas', '-llapack']:
                 ret.append("    " + d)
             ret.append(")\n")
 
